webapp/plugins/vue-directives.js

```diff
@@ -7,7 +7,14 @@ export default ({ app }) => {
       // Focus the element
       Vue.nextTick(() => {
         if (binding.value !== false) {
-          el.focus()
+          let target = el
+          const isInput = target instanceof HTMLInputElement
+          if (!isInput) {
+            target = el.querySelector('input')
+          }
+          if (target && typeof target.focus === 'function') {
+            target.focus()
+          }
```

Comment:
is the fix to the issue you were facing in the tests? do you have a link to where you found this fix?
Also, would need to have a closer look at this.

Reply:
I don't think so.
`v-focus` just existed; it was not used by any component, so I took the liberty to make it target <input\> elements.

Reply:
I meant the issue with `Vue.nextTick()` not doing what we expected in the test you wrote.

Reply:
No, this vue-directives `v-focus` change is unrelated to that.

Like you suggested, I had a look at other use cases of Vue.nextTick() in the tests and noticed that none of those other instances waits for the actual result of a function. It's always just used like
```js
wrapper.find('button').emit('click');
await Vue.nextTick();
expect(getFooAsync).toHaveBeenCalled()
```
i.e. not testing with what the Promise to resolved, merely whether its producing function had been called.

So I went to the [jest documentation](https://jestjs.io/docs/en/timer-mocks) and found that jest manages its own micro-queue for timers.

There is a package called flush-promises which I think is mentioned in the vue docs, but I didn't want to install another package. Turns out, that it is [already in use](https://github.com/Human-Connection/Human-Connection/blob/f13eb29ad342392edee61b82c02a126a5fdbae3b/webapp/pages/settings/my-social-media.spec.js#L64).

Reply:
yes, that's great! nice catch!